fs/vfs/fs_dup2.c

```diff
@@ -107,10 +110,25 @@ int file_dup2(FAR struct file *filep1, FAR struct file *filep2)
 
           temp.f_priv = filep1->f_priv;
 
+          /* Add nonblock flags to avoid happening block when
+           * calling open()
+           */
+
+          temp.f_oflags |= O_NONBLOCK;
+
           if (inode->u.i_ops->open)
             {
               ret = inode->u.i_ops->open(&temp);
             }
+
+          if (ret >= 0 && (filep1->f_oflags & O_NONBLOCK) == 0)
+            {
+              ret = file_ioctl(&temp, FIONBIO, 0);
+              if (ret < 0 && inode->u.i_ops->close)
+                {
+                  ret = inode->u.i_ops->close(&temp);
```

Comment:
What is `ret` will be `0` after `close`? Will we drop error code here and continue a "happy path"?